xds/src/main/java/io/grpc/xds/internal/sds/XdsServer.java

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright 2020 The gRPC Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.grpc.xds.internal.sds;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import com.google.common.annotations.VisibleForTesting;
+import io.grpc.Server;
+import io.grpc.ServerServiceDefinition;
+import io.grpc.xds.XdsClientWrapperForServerSds;
+
+import java.io.IOException;
+import java.net.SocketAddress;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+import javax.annotation.Nullable;
+
+/**
+ * Wraps a {@link Server} delegate and {@link XdsClientWrapperForServerSds} and intercepts
+ * {@link Server#shutdown()} etc to properly shut down the {@link XdsClientWrapperForServerSds}
+ * object.
+ */
+@VisibleForTesting
+public final class XdsServer extends Server {
+  private final Server delegate;
+  @Nullable private final XdsClientWrapperForServerSds xdsClientWrapperForServerSds;
+
+  XdsServer(Server delegate, XdsClientWrapperForServerSds xdsClientWrapperForServerSds) {
```

Comment:
nit: add `@Nullable` to the param xdsClientWrapperForServerSds. but, i don't see why it wants null? if this is null the caller can use delegate Server. factory method returns Server type can easily achieve that if caller doesn't want to check null.